mesonbuild/interpreter.py

```diff
@@ -3972,9 +3978,21 @@ def add_arguments(self, node, argsdict, args_frozen, args, kwargs):
             argsdict[lang] = argsdict.get(lang, []) + args
 
     @noKwargs
-    @noPosargs
     def func_environment(self, node, args, kwargs):
-        return EnvironmentVariablesHolder()
+        if args:
+            FeatureNew('environment positional arguments', '0.52.0').use(self.subproject)
```

Comment:
I would move the FeatureNew into the `len(args) == 1` case, just in case we add more args later. Bonus penguin points if you also add that to `func_configuration_data()` too :D

Reply:
done